sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/RapidsShuffleClient.scala

```diff
@@ -501,18 +558,15 @@ class RapidsShuffleClient(
     * @param shuffleRequests blocks to fetch
     * @param handler iterator to callback to
     */
-  private[this] def doHandleMetadataResponse(tx: Transaction,
-                                             resp: RefCountedDirectByteBuffer,
-                                             shuffleRequests: Seq[ShuffleBlockBatchId],
-                                             handler: RapidsShuffleFetchHandler): Unit = {
+  private[this] def doHandleMetadataResponse(
+      tx: Transaction,
+      resp: RefCountedDirectByteBuffer,
+      shuffleRequests: Seq[ShuffleBlockBatchId],
+      handler: RapidsShuffleFetchHandler): Unit = GpuShuffleEnv.synchronized{
```

Comment:
Why do we need to synchronize here?

Reply:
whoops. I had some non-thread-safe counters I was using to test this in a real job, and forgot to remove this.

Reply:
Removed in https://github.com/NVIDIA/spark-rapids/pull/331/commits/5b0cd201dbb7214891652ea82f2b97880f4aa516